src/Cryptol/Symbolic.hs

```diff
@@ -123,7 +125,7 @@ finType ty =
     TVIntMod n       -> Just (FTIntMod n)
     TVSeq n t        -> FTSeq <$> numType n <*> finType t
     TVTuple ts       -> FTTuple <$> traverse finType ts
-    TVRec fields     -> FTRecord <$> traverse (traverseSnd finType) fields
+    TVRec fields     -> FTRecord <$> traverse (traverseSnd finType) (sortBy (comparing fst) fields)
```

Comment:
I feel like this is the wrong solution. There are various functions that consume these `FinType` things, and apparently some of them go wrong if they encounter an `FTRecord` with unsorted fields. The way you've done it, all those consumer functions are still broken, because we can still construct `FinType` values with unsorted record fields by other means—at the very least the consumer functions have an undocumented precondition.

I would prefer to either
1. fix all the consumer functions to handle unsorted fields correctly, or
2. change `FTRecord` from `[(Ident, FinType)]` to `Map Ident FinType` to enforce the ordering invariant.

Reply:
I feel like whatever we do here is a bit of a stopgap until we solve #706. If we're going to touch a bunch of places in the code and downstream clients too, I think we should go all the way and get it right.

Reply:
Also, FWIW, This is the only use of `FTRecord` as a constructor in the `cryptol`, `cryptol-verifier`, `SAWCore` or `SAWScript` codebases.

Reply:
So we wouldn't actually be breaking anything downstream by changing `FTRecord` to use a `Map`, is that right? The `FinType` datatype is only used internally to the symbolic backends of Cryptol, so I don't see any reason why not to change it to `Map` now.

Reply:
*shrug* yeah, fair enough